syntax = "proto3";

package counter;

option go_package = "counter/counterpb";

// Enable JavaScript code generation
option optimize_for = SPEED;

// Define the service
service CounterService {
  // A simple RPC to get count
  rpc GetCount (CountRequest) returns (CountResponse) {}
}

// Request message - can be empty since we're just getting a count
message CountRequest {
}

// Response message containing the count
message CountResponse {
  int32 count = 1;
}